projects/step-core/src/lib/shared/function-configuration-dialog.form.ts

```diff
@@ -0,0 +1,98 @@
+import { AbstractControl, FormBuilder, ValidationErrors } from '@angular/forms';
+import { DynamicValueInteger, Function as Keyword } from '../client/generated';
+import { KeyValuePair } from '../domain';
+import { AgentTokenSelectionCriteriaForm } from './agent-token-selection-criteria.form';
+import { FunctionType } from './function-type.enum';
+import { dynamicValueFactory, toKeyValuePairs, toRecord } from './utils';
+
+export type FunctionConfigurationDialogForm = ReturnType<typeof functionConfigurationDialogFormCreate>;
+
+export const schemaValidator = (schemaControl: AbstractControl<string>): ValidationErrors | null => {
+  const schema = schemaControl.value;
+
+  if (!schema) {
+    return {
+      required: true,
+    };
+  }
+
+  let schemaFormatValid = true;
+
+  try {
+    JSON.parse(schema);
+  } catch (error) {
+    schemaFormatValid = false;
+  }
+
+  if (!schemaFormatValid) {
+    return {
+      format: true,
+    };
+  }
+
+  return null;
+};
+
+const DEFAULT_SCHEMA_VALUE = '{}';
+const DEFAULT_CALL_TIMEOUT_MS = 180000;
+
+export const functionConfigurationDialogFormCreate = (
```

Comment:
Most of these forms are belong to the concrete components and don's used anywhere else.
From this point of view, they can be considered as the part of the concrete components. I think it make sense to move the to corresponding component's folders.